In [7]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')
df = pd.read_csv('/content/drive/MyDrive/analytics_projects/bank.csv')
print(df.shape) # row x column


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
(11162, 17)


In [8]:
print(df.head()) # sample rows

   age         job  marital  education default  balance housing loan  contact  \
0   59      admin.  married  secondary      no     2343     yes   no  unknown   
1   56      admin.  married  secondary      no       45      no   no  unknown   
2   41  technician  married  secondary      no     1270     yes   no  unknown   
3   55    services  married  secondary      no     2476     yes   no  unknown   
4   54      admin.  married   tertiary      no      184      no   no  unknown   

   day month  duration  campaign  pdays  previous poutcome deposit  
0    5   may      1042         1     -1         0  unknown     yes  
1    5   may      1467         1     -1         0  unknown     yes  
2    5   may      1389         1     -1         0  unknown     yes  
3    5   may       579         1     -1         0  unknown     yes  
4    5   may       673         2     -1         0  unknown     yes  


In [9]:
print(df.dtypes) # data types

age           int64
job          object
marital      object
education    object
default      object
balance       int64
housing      object
loan         object
contact      object
day           int64
month        object
duration      int64
campaign      int64
pdays         int64
previous      int64
poutcome     object
deposit      object
dtype: object


In [10]:
print(df.isnull().sum()) # no of null records in each column

age          0
job          0
marital      0
education    0
default      0
balance      0
housing      0
loan         0
contact      0
day          0
month        0
duration     0
campaign     0
pdays        0
previous     0
poutcome     0
deposit      0
dtype: int64


In [11]:
print(df["deposit"].value_counts()) # details of particular column

deposit
no     5873
yes    5289
Name: count, dtype: int64


In [12]:
print(df["deposit"].value_counts(normalize=True).round(3)) # details of particular column

deposit
no     0.526
yes    0.474
Name: proportion, dtype: float64


In [13]:
print(df.groupby("job")["deposit"].value_counts(normalize=True).round(2)) # details of particular column grouped by another column

job            deposit
admin.         no         0.53
               yes        0.47
blue-collar    no         0.64
               yes        0.36
entrepreneur   no         0.62
               yes        0.38
housemaid      no         0.60
               yes        0.40
management     yes        0.51
               no         0.49
retired        yes        0.66
               no         0.34
self-employed  no         0.54
               yes        0.46
services       no         0.60
               yes        0.40
student        yes        0.75
               no         0.25
technician     no         0.54
               yes        0.46
unemployed     yes        0.57
               no         0.43
unknown        no         0.51
               yes        0.49
Name: proportion, dtype: float64


In [17]:
df["age_group"]= pd.cut(df["age"],bins = [0,30,40,50,100],labels = ["<30","30-40","40-50","50+"])
print(df.groupby("age_group")["deposit"].value_counts(normalize=True).round(3))
print()
print(df.groupby("deposit")["balance"].mean(),round(0))

age_group  deposit
<30        yes        0.571
           no         0.429
30-40      no         0.567
           yes        0.433
40-50      no         0.592
           yes        0.408
50+        yes        0.531
           no         0.469
Name: proportion, dtype: float64

deposit
no     1280.227141
yes    1804.267915
Name: balance, dtype: float64 0


/tmp/ipykernel_5274/3625366258.py:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  print(df.groupby("age_group")["deposit"].value_counts(normalize=True).round(3))


In [25]:
sub_rate = (df["deposit"] == "yes").mean()
print(f"Overall Subscription rate: {sub_rate:0.2%}")

Overall Subscription rate: 47.38%


#KEY FINDINGS
Overall Subscription Rate : 47.4 %
Students have the highest rate of subscription (74.7 %)
Customer with higher balance subscribe at higher rate